app/models/manager_refresh/save_collection/helper.rb

```diff
@@ -50,9 +50,16 @@ def create_or_update_inventory!(inventory_collection, record_index, association)
       unique_index_keys = inventory_collection.manager_ref_to_cols
 
       association.find_each do |record|
-        # TODO(lsmola) the old code was able to deal with duplicate records, should we do that? The old data still can
-        # have duplicate methods, so we should clean them up. It will slow up the indexing though.
-        record_index[inventory_collection.object_index_with_keys(unique_index_keys, record)] = record
+        index = inventory_collection.object_index_with_keys(unique_index_keys, record)
```

Comment:
So the ems_id + ems_ref will always be unique, but we use the bios uuid for the `uid_ems` and that isn't guaranteed to be unique and there isn't anything we can do about that.
Now if we could get away from using `uid_ems` and always use ems_id + ems_ref... :smile: